Greet-Dough/src/main/java/store/relation/SubStoreImpl.java

```diff
@@ -0,0 +1,31 @@
+package store.relation;
+
+import java.util.ArrayList;
+
+// Store subscriptions as ArrayList or HashMap?
+// A single user's subscriptions probably won't be so large that a linear search will take very long
+public class SubStoreImpl extends Relation {
```

Comment:
this looks good - i took a look in IntelliJ at the object hierarchy. 

my recommendation would be to do something like `public class SubStoreImpl extends Relation implements SubStore` and abstract the public methods in your Store classes. this way, when you use a database instead of hashmaps/lists/etc, you can just re-implement the top level interface.

intellij has an "extract interface" refactoring you can use so that you don't have to rewrite everything. in any case, it should be straightforward to do manually as well!

then, in your Server, where you're instantiating the store, you can actually do `SubStore subStore = new SubStoreImpl()`, and the classes that depend on it can also depend on `SubStore` instead of `SubStoreImpl`. down the road, super easy to just change the implementation to `SubStorePostgres` in the Server main method instead of everywhere in the code

hope that makes sense!

Reply:
cc @DerekLeeCS